src/utils/crypto_utils/tests/crypto_utils_ut.cpp

```diff
@@ -165,3 +170,206 @@ TEST_CASE("Signature Verification")
             key));
     }
 }
+
+TEST_CASE("CryptoUtils_InitAndAllocKeyDataFromUrlEncodedB64String")
+{
+    SECTION("Successful Init")
+    {
+        std::string b64EncodedKeyData = "urq6usAMCRI0VnjL2voQAg==";
+
+        std::array<uint8_t, 16> decodedBytes = { 0xBA, 0xBA, 0xBA, 0xBA, 0xC0, 0x0C, 0x09, 0x12,
+                                                 0x34, 0x56, 0x78, 0xCB, 0xDA, 0xFA, 0x10, 0x02 };
+
+        keydata_wrapper key;
+
+        REQUIRE(CryptoUtils_InitAndAllocKeyDataFromUrlEncodedB64String(key.address_of(), b64EncodedKeyData.c_str()));
+
+        CHECK(key.get() != nullptr);
+
+        REQUIRE(key.get()->keyData != nullptr);
+
+        const CONSTBUFFER* keyBytes = CONSTBUFFER_GetContent(key.get()->keyData);
+
+        REQUIRE(keyBytes != nullptr);
+
+        CHECK(keyBytes->size == decodedBytes.size());
+
+        for (int i = 0; i < keyBytes->size; ++i)
+        {
+            CHECK(keyBytes->buffer[i] == decodedBytes.at(i));
+        }
+
+        CryptoUtils_DeAllocKeyData(key.get());
```

Comment:
use RAII.  maybe keydata_wrapper?

Reply:
and below...

Reply:
The pointer itself is protected by RAII. The data within keydata is not. That's what the DeAlloc is for.